net/netxng/inc/TNetXNGFile.h

```diff
@@ -58,7 +58,8 @@ class TNetXNGFile: public TFile {
       fFile(0), fUrl(0), fMode(XrdCl::OpenFlags::None), fInitCondVar(0),
       fReadvIorMax(0), fReadvIovMax(0) {}
    TNetXNGFile(const char *url, Option_t *mode = "", const char *title = "",
-               Int_t compress = 1, Int_t netopt = 0, Bool_t parallelopen = kFALSE);
+               Int_t compress = 1, Int_t netopt = 0, Bool_t parallelopen = kFALSE,
+	       const char *lurl = 0);
    virtual ~TNetXNGFile();
```

Comment:
If I put the new const char* to the end of formal parameter list of the ctor and default it to 0, like I did, I think, all calls to the old ctor without the last parameter automatically call the new ctor, with the last missing parameter's default value. That way it wouldn't break existing code and generate less object code(okay: very minor savings here, that most likely would be optimized out ). Or do you mean, it's better to use lurl as the second parameter for clarity?

Reply:
For compatibility you need a different constructor, setting the new parameter to a default 0 is not enough.
And in the new constructor, you cannot have the new parameter as last argument with default otherwise you'll have compilation problems. Hence the suggestion above.

Reply:
Hm, I will do as you suggested, but I still wonder why. Its not really the right place, but I'd be happy to understand the difference, so I can learn something from it, if you do have the time.
```c++
class Test{
	public:
 Test(char a='1', char b='2', char c='3'){std::cout<<a<<" "<<b<<" "<<c<<std::endl;}
};
---------------------------------------------------------
int main(){
 Test('h','m');
 return 0;
}
```

Compiling and 
running this, the output is

>  h m 3

I thought, all calls to the 'old' constructor would instead just call the modified constructor like this.
Where and why wouldn't old code compile like this? 

Even if you don't have the time, all thanks in advance